### ISSUE TITLE
Question: Leia official integration ?

### ISSUE THREAD

**Issue by mcallegari** [asciidisco/plugin.video.netflix#32]
Hi, first of all thank you very much for the excellent work you're doing here !

I was wondering if you are in contact with the Kodi team for an official integration into Kodi 18.
If so, can you please point me (us) to the reference place where the discussion is happening ?
I'd love to follow the progress and eventually try this addon myself !

Thanks again !

**Comment by asciidisco** [comment]
@mcallegari 
Hi,
first of all, I´m not sure what you mean with "official integration"?!
If you ask if that plugin will be tightly integrated into Kodi itself, then I can point you to this issue: https://github.com/FernetMenta/kodi-agile/issues/172 - Short version: Not going to happen for good reasons.

Your question could also be understood in a way, that you don't need Kodi Agile, that the plugin does run on the official Kodi 18 builds. If that´s your question, then I can say that work in that direction is ongoing & for sure (I´d say in a couple of weeks) this should be the case. Having it work with mainline Kodi is the primary goal.

If you´re asking if the plugin will end up in the Kodi.tv repo. I´m fairly certain, that this won't happen. First of all, you need the widevine library, which you need to somehow integrate into your Kodi installation (which is tricky in terms of licensing), as only official distributors are allowed to ship with their software (the License actually goes that far to say, that even only the `use` of it must be blessed by Google). Second, that plugin has no blessing from Netflix itself, so putting that into the main Kodi repo might get the Kodi team in to another companies crosshair & we all know that they´ve got all hands full with fighting illegal content. So, it´s nit wise to open up another front.

I hope I could answer your question here, whatever its main intention was.

**Comment by mcallegari** [comment]
Hi, thanks for your answer.
Yes, by "official integration" I meant "will run on official Kodi builds".
You need to excuse my ignorance, but it's not clear to me what "Kodi agile" builds are ? Aren't they nightly builds or do they come from a "unofficial" fork of the main repo ?

Where the addon is stored is another story, but reading your answer made me think of libmad.
MP3 licensing is tricky as well, and it would taint the GPL license, but the trick over the years has been (AFAIK):
- the library is available in the Ubuntu repos. Basically someone just built an open source project, packaged it and published it online
- the user is the one who decides if he wants to taint his installation with a software not complying with the base license

So in other words. Ubuntu cannot send it to you. You have to download it and take the responsibility for it. Same goes for proprietary drivers like nVidia, AMD, WiFi stuff, etc. Pretty fucked up, isn't it ? :)

Maybe the same paradigm can be used for the widevine library ?

**Comment by mcallegari** [comment]
P.S. When you're done with the rework you're doing, I am very willing to provide an italian translation !

**Comment by asciidisco** [comment]
@mcallegari Unfortunately it´s a bit more complicated with the widevine library (As if your example wouldn't be complicated enough itself...)
Google strictly says, even the use of it (without their blessing) is a violation of their EULA. We´ve been thinking of many ways how we could download the browser for the OS & extracting that lib or point Inputstream into the right directory, so that we don't violate the EULA by extracting files etc.
But in the end, my contact at google said that this would all be pointless.
It seems that the only way to have it waterproof, is to sign papers with google (which individuals can't, only organisations or companies can do so). We´re investigating what can be done there, and hope to find a clean solution with a fine user experience in the end. Fingers crossed.

P.s.: Thanks for the translation offer, I´ll ping you once I´m done (might be another week or two, as I´m iterating over the complete code base, to make it compliant with python styleguides etc.)

P.p.s.: I hope you don't mind if I close this one, as there is no action that can be taken from our side now.

P.p.p.s.: Nice avatar :D

**Comment by mcallegari** [comment]
> P.s.: Thanks for the translation offer, I´ll ping you once I´m done (might be another week or two, as I´m iterating over the complete code base, to make it compliant with python styleguides etc.)

> P.p.s.: I hope you don't mind if I close this one, as there is no action that can be taken from our side now.

No problem at all

> P.p.p.s.: Nice avatar :D

Get ready fot the 10th ! The doctor returns :)
